Sources/CircularProgress/util.swift

```diff
@@ -30,6 +29,10 @@ extension NSColor {
 			return NSColor(red: 0.10, green: 0.47, blue: 0.98, alpha: 1)
 		}
 	}()
+
+	var desaturated: NSColor {
+		return NSColor.black // TODO: Desaturate color
+	}
```

Comment:
In my private framework, I have this implemented as:

```swift
extension NSColor {
	typealias HSBAColor = (hue: Double, saturation: Double, brightness: Double, alpha: Double)
	var hsba: HSBAColor {
		var hue: CGFloat = 0
		var saturation: CGFloat = 0
		var brightness: CGFloat = 0
		var alpha: CGFloat = 0
		let color = usingColorSpace(.deviceRGB) ?? self
		color.getHue(&hue, saturation: &saturation, brightness: &brightness, alpha: &alpha)
		return HSBAColor(Double(hue), Double(saturation), Double(brightness), Double(alpha))
	}
}

extension NSColor {
	private func colorWithSaturation(ratio: Double) -> NSColor {
		let color = hsba
		return NSColor(
			hue: CGFloat(color.hue),
			saturation: CGFloat(color.saturation * ratio),
			brightness: CGFloat(color.brightness),
			alpha: CGFloat(color.alpha)
		)
	}

	func desaturating(by ratio: Double) -> NSColor {
		return colorWithSaturation(ratio: 1 - ratio)
	}
}
```

Reply:
I have something similar, but I'll use yours. Thanks. 👍

Reply:
Please let me know if your implementation does something better, in your opinion. I'm always looking to improve my utility methods.